### PR TITLE
Make dev server listen host to be configurable

### PR DESCRIPTION
**Pull request by gauravtiwari**
This PR makes dev server listen host configurable:  

Fixes: https://github.com/rails/webpacker/issues/709
Ref: https://github.com/rails/webpacker/issues/572 

Not sure if this approach is better or using an env variable?

### CODE REVIEW

**Comment by javan** [comment]
What's the reason for CLI args generally when you can configure `webpacker.yml`?

**Comment by gauravtiwari** [comment]
Current we have hardcoded listen address: https://github.com/rails/webpacker/blob/master/lib/install/bin/webpack-dev-server.tt#L19

and https://github.com/rails/webpacker/blob/master/lib/install/bin/webpack-dev-server.tt#L60

File: lib/install/bin/webpack-dev-server.tt [hunk 1]
```diff
@@ -47,9 +53,10 @@ rescue Errno::EADDRINUSE
 end
 
 # Delete supplied host and port CLI arguments
-["--host", "--port"].each do |arg|
+["--host", "--port", "--listen-host"].each do |arg|
   ARGV.delete(args(arg))
   ARGV.delete(arg)
+  ARGV.delete(arg)
```
Comment (curvo) [inline lib/install/bin/webpack-dev-server.tt]: Why did you add this line?  Isn't it redundant with the one above it?

File: lib/install/bin/webpack-dev-server.tt [hunk 2]
```diff
@@ -16,13 +16,19 @@ CONFIG_FILE       = File.join(APP_PATH, "config/webpacker.yml")
...
 WEBPACK_CONFIG    = File.join(APP_PATH, "config/webpack/#{NODE_ENV}.js")
 
-LISTEN_HOST_ADDR = NODE_ENV == 'development' ? 'localhost' : '0.0.0.0'
-
 def args(key)
   index = ARGV.index(key)
   index ? ARGV[index + 1] : nil
 end
 
+LISTEN_HOST_ADDR = if args('--listen-host')
```
Comment (curvo) [inline lib/install/bin/webpack-dev-server.tt]: I think this could be cleaner with something like this...

First, just rename the current LISTEN_HOST_ADDR to something like DEFAULT_LISTEN_HOST_ADDR:

```ruby
DEFAULT_LISTEN_HOST_ADDR = NODE_ENV == 'development' ? 'localhost' : '0.0.0.0'
```

Then within the begin and parallel to the other constants like (DEV_SERVER_ADDR) do this:

```ruby
  HOSTNAME        = args('--host') || dev_server["host"]
  PORT            = args('--port') || dev_server["port"]
  HTTPS           = ARGV.include?('--https') || dev_server["https"]
  DEV_SERVER_ADDR = "http#{"s" if HTTPS}://#{HOSTNAME}:#{PORT}"
  LISTEN_HOST_ADDR = args('--listen-host') || DEFAULT_LISTEN_HOST_ADDR
```

File: lib/install/bin/webpack-dev-server.tt [hunk 3]
```diff
@@ -47,9 +53,10 @@ rescue Errno::EADDRINUSE
 end
 
 # Delete supplied host and port CLI arguments
```
Comment (curvo) [inline lib/install/bin/webpack-dev-server.tt]: Might want to update comment as well:

```ruby
# Delete supplied host, port and listen-host CLI arguments
```

File: lib/install/bin/webpack-dev-server.tt (hunk 2 above)
Comment (gauravtiwari) [reply lib/install/bin/webpack-dev-server.tt]: thanks, pushed an update 👍  💯

File: lib/install/bin/webpack-dev-server.tt (hunk 1 above)
Comment (gauravtiwari) [reply lib/install/bin/webpack-dev-server.tt]: haha, don't know - removed it 👍

File: lib/install/bin/webpack-dev-server.tt (hunk 3 above)
Comment (gauravtiwari) [reply lib/install/bin/webpack-dev-server.tt]: Done 👍
